monix-reactive/shared/src/main/scala/monix/reactive/internal/operators/DoOnTerminateOperator.scala

```diff
@@ -71,12 +71,13 @@ private[reactive] final class DoOnTerminateOperator[A](
         }
       }
 
-      private def onFinish(ex: Option[Throwable]): Task[Unit] = {
-        @inline def triggerSignal(): Unit = ex match {
+      private def triggerSignal(ex: Option[Throwable]): Unit =
```

Comment:
Because the inner-method captures the current context, it can do an extra-allocation, since it's probably going to get translated into a class initialized with whatever it captures in its constructor. Moving it outside ensures it's not capturing anything, and invoking it will not create an extraneous object.

Reply:
Compiler has never done that. The desugaring of inner methods is actually pretty much the same as hand-written there. You can see it by doing `-Xprint:lambdalift` (scala2) or `-Xprint:lambdaLift` (scala 3). You can check it easily on Scastie, [e.g. this](https://scastie.scala-lang.org/qwOq9riDRQWSHi6zVJDsRQ)

Reply:
Hmm, my knowledge there may be outdated, goes to show that we should always check and re-check our assumptions.

Will revert the code change.